cli-reference-schema.yaml

```diff
@@ -165,25 +165,42 @@ properties:
                         </ul>
                       default: false
                     type:
-                      type: string
-                      description: |
-                        The parameter type of the positional argument or parameter.
-                        Valid options are:
-                        - str
-                        - bool
-                        - int
-                      x-intellij-html-description: |
-                        The parameter type of the positional argument or parameter.<br>
-                        Valid options are:<br>
-                        <ul>
-                        <li>str</li>
-                        <li>bool</li>
-                        <li>int</li>
-                        </ul>
-                      enum:
-                        - str
-                        - bool
-                        - int
+                      oneOf:
+                        - type: string
+                          enum:
+                            - int
+                            - str
+                            - bool
+                            - size
+                        - type: object
+                          additionalProperties: false
+                          properties:
+                            range:
+                              type: object
+                              required:
+                                - min
+                                - max
+                              properties:
+                                min:
+                                  type: integer
+                                max:
+                                  type: integer
+                        - type: object
+                          additionalProperties: false
+                          properties:
+                            regex:
+                              type: string
+                        - type: object
+                          additionalProperties: false
+                          minProperties: 1
+                          properties:
+                            size:
+                              type: object
+                              properties:
```

Comment:
I feel like min and max on the size type should have regex patterns applied to match the string that can be set as a size descriptor